packages/terra-collapsible-menu-view/src/CollapsibleMenuViewItemGroup.jsx

```diff
@@ -48,10 +48,14 @@ class CollapsibleMenuViewItemGroup extends React.Component {
 
   handleOnChange(event, selectedIndex) {
     if (this.props.onChange) {
-      let selectedKey = '';
+      let selectedKey = selectedIndex;
       React.Children.forEach(this.props.children, (child, index) => {
         if (selectedIndex === index) {
           selectedKey = child.key;
+
+          // React prepends '.$' to the original key of the element when that element is cloned using React.cloneElement.
+          // We should find a better way to handle this but for now, we have to remove '.$' from the key in order to match the original key.
+          selectedKey = selectedKey.replace(/[.$]/g, '');
```

Comment:
Does anyone have any other suggestion to handle this?

Reply:
Unfortunately, I don't. I think we can move forward with this unless someone else can think of something.

Reply:
possibly not helpful but here a github issue about this topic https://github.com/facebook/react/issues/11464

Reply:
It does look like it is React.Children.map that prepends '.$' and not React.cloneElement. There are at least a few places([CollapsibleMenu](https://github.com/cerner/terra-core/blob/master/packages/terra-collapsible-menu-view/src/_CollapsibleMenu.jsx#L88-L102), [SelectableList](https://github.com/cerner/terra-core/blob/master/packages/terra-list/src/SelectableList.jsx#L54-L59), [ButtonGroup](https://github.com/cerner/terra-core/blob/master/packages/terra-button-group/src/ButtonGroup.jsx#L69-L77)) that map these children. @mjhenkes suggested to try to change React.Children.map to React.Children.forEach. forEach does't return the children so we would have to save and overwrite the children. [CollapsibleMenu](https://github.com/cerner/terra-core/blob/master/packages/terra-collapsible-menu-view/src/_CollapsibleMenu.jsx#L88-L102) even maps these children recursively so changing to React.Children.forEach can get a little messy.

Reply:
Thanks @mjhenkes. I've changed from React.Children.map to React.Children.forEach and no longer have the need to remove '.$' from the key. I updated only the places that affected this.

https://github.com/cerner/terra-core/pull/1206/commits/1a90f8bff94f02d225b98cb71af50667ee9b50b5